pkgs/applications/editors/emacs/generic.nix

```diff
@@ -337,6 +340,9 @@ mkDerivation (finalAttrs: (lib.optionalAttrs withNativeCompilation {
   ++ lib.optional withXwidgets "--with-xwidgets"
   ;
 
+  CFLAGS = lib.optional (variant == "macport")
+    "-include ${./macport_noescape_noop.h}";
```

Comment:
Where this file came from?

Reply:
Ah, I read the huge thread. Nonetheless,  it deserves at least an accompanying comment, like 

`fixes behavior on new LLVM`